CHANGELOG.md

```diff
@@ -35,6 +35,7 @@ should change the heading of the (upcoming) version to include a major version b
 - Fix case where NumberField would not properly reset the field when using programmatic form reset (#4202)[https://github.com/rjsf-team/react-jsonschema-form/issues/4202]
 - Updated widgets to handle undefined `target` in `onFocus` and `onBlur` handlers
 - Fix field disable or readonly property can't cover globalOptions corresponding property (#4212)[https://github.com/rjsf-team/react-jsonschema-form/pull/4212]
+- Fixed `omitExtraData` not working in `onSubmit` and `validateForm`; fixing [#4187](https://github.com/rjsf-team/react-jsonschema-form/issues/4187), [#4165](https://github.com/rjsf-team/react-jsonschema-form/issues/4165) and [#4109](https://github.com/rjsf-team/react-jsonschema-form/issues/4109)
 
```

Comment:
@helen-m-lin Since we released `5.18.5`, can you move this change comment up to the `5.18.6`

Reply:
I tried rebasing your PR already and then I realized your comment needed to move. Plus another fix just merged so you'll have to rebase again.

Reply:
@heath-freenome I moved the changelog comment to `5.18.6`. Let me know if there is anything else!